tests/test_dimension.py

```diff
@@ -830,6 +831,69 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_condition_lowering(self):
+        """
+        Test the lowering of a ConditionalDimension's condition.
+        This test makes an Operator that should indexify and lower the condition
+        passed in the Conditional Dimension
+        """
+
+        grid = Grid(shape=(3, 3))  # Define grid
+
+        # Define Function
+        g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+        g.data[:] = 1
+        x, y = grid.dimensions
+
+        ci = ConditionalDimension(name='ci', parent=y, condition=CondEq(g[x, y], 1))
+
+        f = Function(name='f', shape=grid.shape, dimensions=(x, ci))
+        Operator(Eq(f[x, ci], g[x, y])).apply()
+        assert np.all(f.data == 1)
+
+    def test_condition_subdomains(self):
```

Comment:
NOTE FOR REVIEWERS: This test can be deleted later, I just leave it temporarily here
to check the output of this script.